src/Turing.jl

```diff
@@ -119,6 +114,15 @@ Sampler(alg) = Sampler(alg, Selector())
 Sampler(alg, model::Model) = Sampler(alg, model, Selector())
 Sampler(alg, model::Model, s::Selector) = Sampler(alg, model, s)
 
+abstract type AbstractContext end
+struct DefaultContext <: AbstractContext end
+struct LikelihoodContext <: AbstractContext end
+struct BatchContext{Tctx, T} <: AbstractContext
```

Comment:
Maybe consider 

1) renaming `BatchContext` to `MiniBatchContext`, since batched methods often refer to algorithms utilising the full dataset? 
2) adding a comment here, explaining the purpose of `BatchContext`?

Reply:
cc @willtebbutt @trappmartin

Reply:
Sure